### PR TITLE
Implement the RYY gate

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Implements the RYY gate, which is frequently used in chemistry for the so-called `SwapRZ` ansatz which preserves the total number of `1`s of the initial state.

### Details and comments

This implementation is up to a phase of `exp(iθ/2)`, similar to the `RXXGate` which also carries a phase.

### CODE REVIEW

File: qiskit/extensions/standard/ryy.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from qiskit.circuit import Gate
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit import QuantumRegister
+
+
+class RYYGate(Gate):
+    """Two-qubit YY-rotation gate.
+
+    This gate corresponds to the rotation U(θ) = exp(-iθ/2 Y⊗Y), multiplied with
+    the phase exp(iθ/2).
```
Comment (mtreinish) [inline qiskit/extensions/standard/ryy.py]: I would probably use sphinx math mode for this part of the docstring

File: qiskit/extensions/standard/__init__.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2017.
+# (C) Copyright IBM 2020.
```
Comment (mtreinish) [inline qiskit/extensions/standard/__init__.py]: I don't think this is correct to do. The copyright date of 2017 is when the module was written adding one line doesn't change that, You can probably add 2020 here or do a range.

File: qiskit/extensions/standard/ryy.py (hunk 1 above)
Comment (Cryoris) [reply qiskit/extensions/standard/ryy.py]: Does sphix math support latex notation? Meaning 
```rst
This gate corresponds to the rotation :math:`U(\theta) = e^{-i\theta / 2 Y \otimes Y)`, 
multiplied with the phase :math:`e^{i \theta / 2}`
```
would be the correct way?

File: qiskit/extensions/standard/__init__.py (hunk 2 above)
Comment (Cryoris) [reply qiskit/extensions/standard/__init__.py]: Yes, you're right. I was always confused about how this is correctly done. Do you know whether either of `2017-2020` or `2017, 2020` is more correct? Currently, most codes I've seen in Qiskit use the comma-separation.

Comment (chriseclectic) [reply qiskit/extensions/standard/__init__.py]: @nonhermitian can confirm policy, but I believe the when editing files you leave the original date.
and new files you use `(C) Copyright IBM 2017, YEAR FILE ADDED`

File: qiskit/extensions/standard/ryy.py (hunk 1 above)
Comment (chriseclectic) [reply qiskit/extensions/standard/ryy.py]: Yes you can put any standard latex in `:math:`. If you use backslashes though you either need to make the doc a raw string `r"""` or use a double backslash for all tex commands.

Comment (Cryoris) [reply qiskit/extensions/standard/ryy.py]: Ok, thanks!

File: qiskit/extensions/standard/__init__.py (hunk 2 above)
Comment (Cryoris) [reply qiskit/extensions/standard/__init__.py]: Okay, thanks for the clarification. I reverted the years to what had been in place previously. 👍

File: qiskit/extensions/standard/ryy.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        from qiskit.extensions.standard.u1 import U1Gate
+        from qiskit.extensions.standard.rx import RXGate
+        definition = []
+        q = QuantumRegister(2, 'q')
+        theta = self.params[0]
+        rule = [
+            (RXGate(np.pi / 2), [q[0]], []),
+            (RXGate(np.pi / 2), [q[1]], []),
+            (CXGate(), [q[0], q[1]], []),
+            (U1Gate(theta), [q[1]], []),
```
Comment (chriseclectic) [inline qiskit/extensions/standard/ryy.py]: I think if you change this to a RzGate it would be phase correct wrt to the canonical RzGate matrix.

**Pull request merged**